### ISSUE TITLE
Refactor: use `Literal` in type hints

### ISSUE THREAD

**Issue by JohannesMessner** [docarray/docarray#1670]

**Comment by srini047** [comment]
Hey, @JohannesMessner I would like to work on this issue.

**Comment by JoanFM** [comment]
hey @srini047 ,

It would be great if you can take up this issue.

**Comment by bpshaver** [comment]
Is `protobuf-array` also a possible value for `protocol` in the intended context? (If not... maybe the parameter name `protocol` should be used to refer to two different things.)

**Comment by bpshaver** [comment]
Opened a [PR](https://github.com/docarray/docarray/pull/1827).

I'm in favor of an enum over a `Literal` consisting of different strings, but I can see how the latter is easier for end users. I can seek out other instances of this same issue if this PR looks good.

**Comment by JoanFM** [comment]
Yes, it would be nice to make sure there are other instances or if we can close this

**Comment by bpshaver** [comment]
I believe these parameters are candidates for similar treatment:

- [ ] `audio_codec`

- [ ]`audio_format`

- [ ] `video_codec`

But there may be more
